src/components/TagCloud/TagCloud.js

```diff
@@ -0,0 +1,26 @@
+import React, { PropTypes } from 'react'
+import {Tag} from '../Tag'
+
+import {styles} from './styles'
+
+const TagCloud = ({names, onClickAction}) => (
+  <div>
+    {
+      names.map((name, i) => (
+       <div {...styles.item}>
+         {onClickAction
+           ? <Tag key={i} name={name} onClickAction={(name) => onClickAction(name)} />
```

Comment:
the list of names supplied to the component will change, so it will be removed from the array that supplies the content to the cloud